.github/workflows/ci.yml

```diff
@@ -33,6 +39,12 @@ jobs:
     - name: Checkstyle
       run: mvn -B checkstyle:checkstyle
 
+    - name: Package
+      run: mvn -B clean package -Padvanced -DskipTests
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
+
```

Comment:
Any thoughts on moving this after test as in original code? Just keeping the flow as  test followed by package (in case test fail early, package can be skipped).

Reply:
The `package` goal is required for CodeQL Analysis and it is better to skip tests as the longest step should the analysis fail. In any case, package just creates .jar and .zip files that would be discarded should test fail.